### ISSUE TITLE
Issues with NotAction policies

### ISSUE THREAD

**Issue by codybuell** [duo-labs/parliament#153]
Checking NotAction policies fails. As a test policy:

```json
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "NotAction": [
        "cloudformation:Delete*",
        "cloudformation:Preview*"
      ],
      "Effect": "Deny",
      "Resource": "*"
    }
  ]
}
```

Checking the above will return:

```sh
$ parliament --file test.json
LOW - Unknown action -  - Unknown action cloudformation:Preview* - {'string': 'cloudformation:Preview*', 'line': 7, 'column': 9, 'filepath': 'test.json'}
```

Removing the unknown action then results in:

```sh
Traceback (most recent call last):
  File "/usr/local/bin/parliament", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/parliament/cli.py", line 316, in main
    policy = analyze_policy_string(
  File "/usr/local/lib/python3.8/site-packages/parliament/__init__.py", line 69, in analyze_policy_string
    policy.analyze(
  File "/usr/local/lib/python3.8/site-packages/parliament/policy.py", line 289, in analyze
    self.check_for_bad_patterns()
  File "/usr/local/lib/python3.8/site-packages/parliament/policy.py", line 197, in check_for_bad_patterns
    refs[priv] = self.get_allowed_resources("s3", priv)
  File "/usr/local/lib/python3.8/site-packages/parliament/policy.py", line 148, in get_allowed_resources
    all_references = self.get_references(privilege_prefix, privilege_name)
  File "/usr/local/lib/python3.8/site-packages/parliament/policy.py", line 84, in get_references
    stmt_relevant_resources = stmt.get_resources_for_privilege(
  File "/usr/local/lib/python3.8/site-packages/parliament/statement.py", line 341, in get_resources_for_privilege
    if not self.in_actions(privilege_prefix, privilege_name):
  File "/usr/local/lib/python3.8/site-packages/parliament/statement.py", line 319, in in_actions
    for action_struct in expand_action(action, raise_exceptions=False):
  File "/usr/local/lib/python3.8/site-packages/parliament/__init__.py", line 189, in expand_action
    parts = action.split(":")
  File "/usr/local/lib/python3.8/site-packages/jsoncfg/config_classes.py", line 230, in __getattr__
    raise JSONConfigNodeTypeError(
jsoncfg.config_classes.JSONConfigNodeTypeError: Expected a ConfigJSONObject but found ConfigJSONScalar. You are trying to get an item from a scalar as if it was an object. item=split [line=6;col=17]
```

Also if the policy contains more than one issue, parliament is only returning one at a time.

This is on parliament version 1.2.1.

**Comment by 0xdabbad00** [comment]
The exception is a bug that should be fixed.


> Also if the policy contains more than one issue, parliament is only returning one at a time.
Parliament is capable of finding multiple issues, but in many circumstances, it will basically bail once it runs into certain types of issues because it is not smart enough to know how to proceed checking further.

**Comment by codybuell** [comment]
Looks like [this commit](https://github.com/duo-labs/parliament/commit/c2d8ff512cd2fa89a84d1829e0c6f73cf0521dbe) addresses the bug, but it hasn't made its way into release yet.

**Comment by 0xdabbad00** [comment]
This is resolved via #151 which has been merged for version 1.2.2.
